### DISCUSSION TITLE
Share Your MQTTX Experience: We Value Your Feedback!

### DISCUSSION THREAD

**Discussion by ysfscream** [emqx/MQTTX#770] [General]
Hello MQTTX Users,

Thank you for being a part of our growing community. We are eager to hear about your experiences with MQTTX!

Your valuable feedback will help us in two significant ways:
- **Showcase:** Selected positive feedback will be featured on our official website to help others learn and get inspired about MQTTX.
- **Improvement:** We are committed to continuously enhancing MQTTX, and your constructive criticism will guide us in identifying areas for improvement. We will incorporate the feedback into our task tracking for timely improvements.

As a token of our appreciation, we will credit your contribution by displaying your GitHub account or company link and name alongside your comments.

To make it easier for you, we have prepared a simple template that you can copy and paste to share your feedback:

```
### GitHub or Company: (please fill in)

### 🌟 Positive Aspects:

### 🚧 Areas for Improvement:
```

Thank you for helping us make MQTTX better for everyone. We look forward to hearing from you!

**Comment by programmer131** [comment]
#### 👍Good
I'm using MQTT X from couple of months now, and starting from 1.7.0 release, this client software is much stable as compared to previous builds, it still consume lot of RAM when few thousand messages appear in history, but thank you for fixing history clean up feature. 
Just wanted to say thank you too all the developers involved in this project.

**Reply by ysfscream** [reply]
Thanks for using and giving us feedback. I should be the one to thank you for your support! : )

And the memory problem we will continue to optimize in the future.

**Comment by rpiper** [comment]
I've been using MQTTX for a few weeks now, and I love it! I really like the modern UI design, and Dark Mode. I haven't run into any problems yet, I only wish it had a Sparkplug protocol decoder built-in. I was previously using an old free version of MQTT.fx, which will decode Sparkplug payloads into a pretty JSON format to inspect the payload contents.

I'm going to look into setting up a build environment so I might be able to help contribute by implementing this Sparkplug decoder feature. Please let me know if I can help.

**Reply by ysfscream** [reply]
Hi, thank you very much for your interest in MQTTX!

Support for Sparkplug is actually always available in Roadmap, but it hasn't been implemented yet, want to use the plugin to support the Sparkplug protocol. This can be viewed in the product roadmap linked below. 👇

https://www.emqx.com/en/blog/mqttx-v-1-7-2-release-notes

If there are any questions we can continue the discussion here.

For JSON format, you can use the right-click menu to choose to beautify the format or set the incoming data to JSON, which will automatically beautify the JSON format

![image](https://user-images.githubusercontent.com/21299158/152713622-908c2cf0-1901-4d83-942e-3fb2ef367a92.png)

![image](https://user-images.githubusercontent.com/21299158/152713637-08ae3629-3be4-4011-b4a0-acce236a53b3.png)

**Comment by awesomezed** [comment]
**How to turn ON and OFF a Relay using ESP8266 and MQTTX and MQTT Broker**. Please help me!!!!!!

**Reply by ysfscream** [reply]
Hi, we could discuss it here: https://github.com/emqx/MQTTX/discussions/885

**Comment by interns-rtsd** [comment]
how do i get/download the source code ? please help me out !!!!!

**Reply by ysfscream** [reply]
On the code page. 😊

**Comment by rsnbusa** [comment]
Hi Everyone. Great product! congrats. May I suggest that certain functions in the right menu ... ,like clear history, could be a button like disconnect/mqtt copilot/edit so that its fast/easy to clear history. Otherwise its rather cumbersome to hit the ... then scroll down to clear history. There is certainly space in that "menubar". Regards
